src/views/ManageList.jsx

```diff
@@ -1,7 +1,126 @@
-export function ManageList() {
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+import toast, { Toaster } from 'react-hot-toast';
+
+const purchaseTimelines = {
+	soon: 7,
+	kindOfSoon: 14,
+	notSoon: 30,
+};
+
+export function ManageList({ listPath }) {
+	const [itemName, setItemName] = useState('');
+	const [itemNextPurchaseTimeline, setItemNextPurchaseTimeline] = useState('');
+
+	const handleItemNameTextChange = (e) => {
+		setItemName(e.target.value);
+	};
+
+	const handleNextPurchaseChange = (e) => {
+		setItemNextPurchaseTimeline(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		const daysUntilNextPurchase = purchaseTimelines[itemNextPurchaseTimeline];
+
+		try {
+			await toast.promise(
+				addItem(listPath, {
+					itemName,
+					daysUntilNextPurchase,
+				}),
+				{
+					pending: 'Adding item to list.',
+					success: () => {
+						setItemName('');
+						setItemNextPurchaseTimeline('');
+						return `${itemName} successfully added to your list!`;
+					},
+					error: () => {
+						return `${itemName} failed to add to your list. Please try again!`;
+					},
+				},
+				{
+					style: {
+						minWidth: '250px',
+					},
+				},
+			);
+		} catch (error) {
+			console.error('Failed to add item:', error);
+		}
+	};
+
 	return (
-		<p>
-			Hello from the <code>/manage-list</code> page!
-		</p>
+		<div>
+			<p>
+				Hello from the <code>/manage-list</code> page!
+			</p>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="item-name">
+					Item:
+					<input
+						id="item-name"
+						type="text"
+						name="item"
+						value={itemName}
+						onChange={handleItemNameTextChange}
+						required
+						aria-label="Enter the item name"
+					/>
+				</label>
+				<br />
+				<fieldset>
+					<legend>When to buy:</legend>
+					<label htmlFor="soon">
+						<input
+							type="radio"
+							id="soon"
+							name="when-to-buy"
+							value="soon"
+							required
+							onChange={handleNextPurchaseChange}
+							checked={itemNextPurchaseTimeline === 'soon'}
+							aria-label="Set buy to soon, within 7 days"
+						/>
+						Soon -- Within 7 days!
```

Comment:
In order for us to obtain the flexibility of `purchaseTimelines`, we'd have to reference that here (and in the aria-label) with something like this for the label text:

> "{ `Soon -- Within ${purchaseTimelines["soon"]}!` }"

And something like this for the aria-label

> `Set buy to soon, within ${purchaseTimelines["soon"]} days`

Reply:
Otherwise, we can't change the values in `purchaseTimelines` without making changes throughout this file 😄